src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -35,13 +35,20 @@ public class AuthorizationCheckService {
 
     private final CourseRepository courseRepository;
 
-    // TODO: rename to course-enrollment
+    @Deprecated(forRemoval = true)
     @Value("${artemis.user-management.course-registration.allowed-username-pattern:#{null}}")
+    private Optional<Pattern> allowedCourseRegistrationUsernamePattern;
+
+    @Value("${artemis.user-management.course-enrollment.allowed-username-pattern:#{null}}")
     private Optional<Pattern> allowedCourseEnrollmentUsernamePattern;
 
     public AuthorizationCheckService(UserRepository userRepository, CourseRepository courseRepository) {
         this.userRepository = userRepository;
         this.courseRepository = courseRepository;
+
+        if (allowedCourseEnrollmentUsernamePattern == null) {
```

Comment:
Did you check that in case of a missing value this is actually null and not an empty optional?
If so, should we make the new variable directly a Pattern instead of an optional?

Reply:
Thx, I must have rolled back one commit too many